### PR TITLE
ui-rex-connect-by-ip

### PR DESCRIPTION
**Pull request by pondrejk**
adding by ip versions of the ui remote execution tests and changing the provisioning server
related tp #4624

### CODE REVIEW

**Comment by sghai** [comment]
@pondrejk: What is holding this PR ? I believe w/ IP you should be able to communicate w/ host. Assuming infra is not blocking the PR ?

**Comment by oshtaier** [comment]
@pondrejk ping

**Comment by oshtaier** [comment]
please rebase and provide test results

**Comment by rplevka** [comment]
this will probably fail without a proper infrastructure.
We're in the process of making `libvirt1` CR to  support `vm.py` provisioning

**Comment by pondrejk** [comment]
This was waiting on #5391 now ready for review, added the _by_ip versions of ui rex tests, skipped the _by_hostname versions due to infra limitations

```
pytest tests/foreman/ui/test_remoteexecution.py -k _by_ip 
========================================== test session starts ==========================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.5.2, pluggy-0.4.0
rootdir: /home/pondrejk/Documents/robottelo, inifile:
plugins: xdist-1.19.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 22 items                                                                                       
2017-12-05 12:24:07 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui/test_remoteexecution.py ....


========================================== 18 tests deselected ==========================================
=============================== 4 passed, 18 deselected in 498.28 seconds ===============================
```

**Comment by san7ket** [comment]
@pondrejk I believe since #5391 is merged the , infra is not a issue anymore

**Comment by oshtaier** [comment]
@pondrejk 
Yeah, I didn't like the idea of stubs here in the first place. Could you please check whether Sanket is right?

**Comment by pondrejk** [comment]
@oshtaier @san7ket  Unfortunately #5391 doesn't help here, it fixed a subnet issue rex tests had but we still are not set up for rex by hostname. Checked with Roman and the problem (lack of DNS capsule) persists. Also if we look at tier3 automation results, we see those ui by-hostname tests keep failing (whereas cli by-ip tests are green), that's why I want to stub those tests and replace them with by-ip alternatives (as it currently is in cli tests), at least to free up resources for now

**Comment by rplevka** [comment]
what @pondrejk said, we don't use the satellite configured to provide dns (infra problem).
Apart from that, the tests are identical.

File: tests/foreman/ui/test_remoteexecution.py [hunk 1]
```diff
@@ -823,3 +827,279 @@ def test_positive_run_job_against_multiple_provisioned_hosts(self):
...
+                    Host.set_parameter({
+                        'host': vm.hostname,
+                        'name': 'remote_execution_connect_by_ip',
+                        'value': 'True',
+                    })
+                with Session(self.browser) as session:
+                    set_context(session, org=self.organization.name)
+                    self.hosts.update_host_bulkactions(
+                        [client.hostname, client2.hostname],
+                        action='Run Job',
```
Comment (rplevka) [inline tests/foreman/ui/test_remoteexecution.py]: This seem to be outdated - it currently fails with:
'element <|strategy=xpath|value=//li/a[text()='Run Job']|> was not found while trying to click'
the actual text is now "schedule remote job":
img

shall this be addressed in different issue&&pr?

Comment (oshtaier) [reply tests/foreman/ui/test_remoteexecution.py]: so how does test pass?

Comment (oshtaier) [reply tests/foreman/ui/test_remoteexecution.py]: oh, we access element by id, so probably it was not changed

Comment (oshtaier) [reply tests/foreman/ui/test_remoteexecution.py]:
```
                if action == 'Run Job':
                    self.assign_value(
                        locators['job_invocation.command'],
                        parameter['command']
                    )
```

 locators['job_invocation.command'] = "job_invocation.command": (By.XPATH, "//textarea[@id='command']"),

where did you find `'element <|strategy=xpath|value=//li/a[text()='Run Job']|>`?

Actually, I can't find such locator in any file

Comment (rplevka) [reply tests/foreman/ui/test_remoteexecution.py]: @oshtaier,
I think it's hidden rather here: (few lines above in `update_host_bulkactions`):
```python
268: self.click(locators['host.select_action'] % action)
```
which boils down to (`locators/base.py`):
```python
268: "host.select_action": (By.XPATH, "//li/a[text()='%s']")
```


> so how does test pass?

this is a million dollar question - @pondrejk are you sure you didn't use `satellite 6.2` or something?
anyway, I spot this error in the test which uses the hostname in the current automation. I just assume the `_by_ip` equivalent does all the same

**Comment by oshtaier** [comment]
@pondrejk ping
according to Friday meeting, we have to leave one test case for hostname and all other be implemented using ip

**Comment by pondrejk** [comment]
ok guys, I reduced the by_hostname tests to 1 and included the amendments from #5646. That locator issue is a mystery to me, though it's possible I went with 6.2 as I work on that simultaneously, apologies, fixed now.

**Comment by rplevka** [comment]
I think this is now addressed and we can safely merge this

**Comment by oshtaier** [comment]
Isn't
```
-                with Session(self) as session:
 +                with Session(self.browser) as session:
                      set_context(session, org=self.organization.name)
                      self.hosts.update_host_bulkactions(
                          [client.hostname, client2.hostname],
                         action='Run Job',
                         parameters_list=[{'command': 'ls'}],
                     )
```
going to fail?

Please, do not merge code without test results

**Pull request merged**